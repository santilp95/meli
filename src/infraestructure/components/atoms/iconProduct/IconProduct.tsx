"use client";
import React from 'react'
import Image from 'next/image';

const logo = "/assets/Logo_ML.png";

/**
 * Icon product component
 */
export const IconProduct = () => {
  return (
    <Image src={logo} alt="Logo" width={40} height={40} />
  )
}
