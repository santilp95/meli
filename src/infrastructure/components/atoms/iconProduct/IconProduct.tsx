"use client";
import { FC } from "react";
import Image from "next/image";

const logo = "/assets/Logo_ML.png";

interface IconProductProps {
  /**
   * Function to handle the click event
   */
  onClick?: () => void;
}

/**
 * Icon product component
 */
export const IconProduct: FC<IconProductProps> = ({ onClick }) => {
  return (
    <button
      onClick={onClick}
      style={{
        border: "none",
        background: "none",
        padding: 0,
        cursor: "pointer",
      }}
      data-testid="logo-icon"
    >
      <Image src={logo} alt="Logo" width={40} height={30} />
    </button>
  );
};
