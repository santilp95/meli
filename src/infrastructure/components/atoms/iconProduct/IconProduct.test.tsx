import {render} from  '@testing-library/react';

import { IconProduct } from './IconProduct';

describe('IconProduct', () => {
  it('should render', () => {
    const { container } = render(<IconProduct />);
    expect(container).toMatchSnapshot();
  });

  it('renders the logo image', () => {
    const { getByAltText } = render(<IconProduct />);
    const logoImage = getByAltText('Logo');
    expect(logoImage).toBeInTheDocument();
    expect(logoImage).toHaveAttribute('src', '/_next/image?url=%2Fassets%2FLogo_ML.png&w=96&q=75');
    expect(logoImage).toHaveAttribute('width', '40');
    expect(logoImage).toHaveAttribute('height', '30');
  });
});