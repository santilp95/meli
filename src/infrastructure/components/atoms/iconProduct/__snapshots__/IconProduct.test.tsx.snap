// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconProduct should render 1`] = `
<div>
  <button
    class="icon-product"
    data-testid="logo-icon"
  >
    <img
      alt="Logo"
      data-nimg="1"
      decoding="async"
      height="30"
      loading="lazy"
      src="/_next/image?url=%2Fassets%2FLogo_ML.png&w=96&q=75"
      srcset="/_next/image?url=%2Fassets%2FLogo_ML.png&w=48&q=75 1x, /_next/image?url=%2Fassets%2FLogo_ML.png&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </button>
</div>
`;
